test/jdk/java/util/regex/RegExTest.java

```diff
@@ -4557,12 +4557,36 @@ public static void badIntersectionSyntax() {
         assertTrue(e.getMessage().contains("Bad intersection syntax"));
     }
 
+    //This test is for 8281560
+    @Test
+    public static void prematureHitEndInNFCCharProperty() {
+        var testInput = "a1a1";
+        var pat1 = "(a+|1+)";
+        var pat2 = "([a]+|[1]+)";
+
+        var matcher1 = Pattern.compile(pat1).matcher(testInput);
+        var matcher2 = Pattern.compile(pat2).matcher(testInput);
```

Comment:
Hello Ian,
The JBS issue notes that the bug is only noticed when the `Pattern.CANON_EQ` is used. The javadoc of this `CANON_EQ` states that this isn't enabled by default. Do you think this test should also include matchers which use Pattern(s) with this `CANON_EQ` enabled?

Reply:
Oh wow yes. Thanks for this catch. I was rewriting the tests to fit this mold and left the flags out. Added them back in. Thanks again!

Reply:
Thank you Ian, the changed version of the test looks good.